diff --git a/include/drm-uapi/drm.h b/include/drm-uapi/drm.h
index ceb0f9e..1b5e2ca 100644
--- a/include/drm-uapi/drm.h
+++ b/include/drm-uapi/drm.h
@@ -45,7 +45,7 @@ typedef unsigned int drm_handle_t;
 
 #include <stdint.h>
 #include <sys/types.h>
-#if defined(__GNU__)
+#if defined(__GNU__) || defined(__EMBOX__)
 #include <sys/ioctl.h>
 #else
 #include <sys/ioccom.h>
diff --git a/src/gallium/auxiliary/hud/hud_context.c b/src/gallium/auxiliary/hud/hud_context.c
index cc43d3b..ab33203 100644
--- a/src/gallium/auxiliary/hud/hud_context.c
+++ b/src/gallium/auxiliary/hud/hud_context.c
@@ -80,7 +80,7 @@ static int hud_scale = HUD_DEFAULT_SCALE;
 static int hud_rotate = HUD_DEFAULT_ROTATION;
 static float hud_opacity = HUD_DEFAULT_OPACITY / 100.0f;
 
-#if DETECT_OS_POSIX
+#if DETECT_OS_POSIX && !DETECT_OS_EMBOX /* no signal delivery at EL1 */
 static void
 signal_visible_handler(int sig, siginfo_t *siginfo, void *context)
 {
@@ -1984,7 +1984,7 @@ hud_create(struct cso_context *cso, struct hud_context *share,
    }
    const char *env = debug_get_option("GALLIUM_HUD",
       emulate_libgl_show_fps ? "stdout,fps" : NULL);
-#if DETECT_OS_POSIX
+#if DETECT_OS_POSIX && !DETECT_OS_EMBOX /* no signal delivery at EL1 */
    unsigned signo = debug_get_num_option("GALLIUM_HUD_TOGGLE_SIGNAL", 0);
    static bool sig_handled = false;
    struct sigaction action;
@@ -2078,7 +2078,7 @@ hud_create(struct cso_context *cso, struct hud_context *share,
    list_inithead(&hud->pane_list);
 
    /* setup sig handler once for all hud contexts */
-#if DETECT_OS_POSIX
+#if DETECT_OS_POSIX && !DETECT_OS_EMBOX /* no signal delivery at EL1 */
    if (!sig_handled && signo != 0) {
       action.sa_sigaction = &signal_visible_handler;
       action.sa_flags = SA_SIGINFO;
